src/build.d

```diff
@@ -1034,7 +1035,9 @@ void processEnvironment()
     }
     if (env.getNumberedBool("ENABLE_RELEASE"))
     {
-        dflags ~= ["-O", "-release", "-inline"];
+        dflags ~= ["-O", "-inline"];
+        if (!env.getNumberedBool("ENABLE_ASSERTS"))
+            dflags ~= ["-release"];
```

Comment:
One thing that we shouldn't forget is that `-release` might generate different code with DMD, so it might be worthwhile to do this only for certain CIs.

Additionally, to tackle this problem on newer CIs there's [`-check` switch](https://dlang.org/dmd-linux.html#switch-check), so we conditionally on the host compiler version we could do `-release -check=assert=on`.